Installation on Windows fails with MAKEFLAGS=-j4

On Windows, if the user's `~/.Renviron` file has `MAKEFLAGS=-j4`, then installation will fail. (The same is probably true if the system Renviron file has that setting.)

The error is:

```
c:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.0/include" -DNDEBUG -I./lib -I../windows/openssl-1.0.2o/include -D_WEBSOCKETPP_CPP11_THREAD_ -I"C:/Users/winst/R/3.5/Rcpp/include" -I"C:/Users/winst/R/3.5/BH/include" -I"C:/Users/winst/R/3.5/AsioHeaders/include"        -O2 -Wall  -mtune=generic -O0 -Wall -c websocket.cpp -o websocket.o
In file included from C:/Users/winst/R/3.5/AsioHeaders/include/asio/ssl/context_base.hpp:19:0,
                 from C:/Users/winst/R/3.5/AsioHeaders/include/asio/ssl/context.hpp:23,
                 from C:/Users/winst/R/3.5/AsioHeaders/include/asio/ssl.hpp:18,
                 from ./lib/websocketpp/common/asio_ssl.hpp:34,
                 from ./lib/websocketpp/transport/asio/security/tls.hpp:35,
                 from ./lib/websocketpp/config/asio_client.hpp:33,
                 from client.hpp:5,
                 from websocket.cpp:24:
C:/Users/winst/R/3.5/AsioHeaders/include/asio/ssl/detail/openssl_types.hpp:20:26: fatal error: openssl/conf.h: No such file or directory
 #include <openssl/conf.h>
                          ^
compilation terminated.
make: *** [C:/PROGRA~1/R/R-35~1.0/etc/i386/Makeconf:215: websocket.o] Error 1
make: *** Waiting for unfinished jobs....
ERROR: compilation failed for package 'websocket'
```

If the `MAKEFLAGS=-j4` is commented out, then installation works fine.

This happens because in the Makevars.win, it tries to run `../tools/winlibs.R`, which downloads the openssl files, at the same time as it compiles the code for websocket. By the time it compiles websocket.cpp, it hasn't yet completed downloading the openssl files, and so the error occurs.

Note that if the `windows/` directory exists (because the winlibs.R script has already run), this problem won't happen. To reproduce the problem, remove the windows/ directory, or just do `install_gitub()`.

--- a/src/Makevars.win
+++ b/src/Makevars.win
@@ -5,7 +5,13 @@ PKG_LIBS= -L../windows/openssl-$(OPENSSL_VERSION)/lib${R_ARCH} -lssl -lcrypto -l
 
 CXX_STD=CXX11
 
-all: clean winlibs
+.PHONY: all winlibs
+
+all: $(SHLIB)
+
+# Need to make sure that openssl is fully downloaded before trying to compile
+# websocket.cpp. See https://github.com/rstudio/websocket/issues/43
+$(OBJECTS): winlibs
 
 clean:
 	rm -f $(SHLIB) $(OBJECTS)

